nextmv/cloud/package.py

```diff
@@ -0,0 +1,370 @@
+            pattern = pattern[1:]
+            negated = True
+        matches = glob.glob(pattern)
+        if not matches and not negated:
+            missing.append(filter)
+        else:
+            if negated:
+                found = [f for f in found if f not in matches]
+            else:
+                found.extend(matches)
+
+    # Switch back to the original directory
+    os.chdir(cwd)
+
+    files = []
+    for file in found:
+        files.append(
+            {
+                "interior_path": file,
+                "absolute_path": os.path.join(app_dir, file),
+            }
+        )
+
+    return found, missing, files
+
+
+def __confirm_mandatory_files(manifest: Manifest, present_files: List[str]) -> None:
+    """Confirm that all mandatory files are present in the given list of files."""
+
+    mandatory_files = _MANDATORY_FILES_PER_TYPE[manifest.type]
+    found_files = {os.path.normpath(file): True for file in present_files}
+    missing_files = [file for file in mandatory_files if file not in found_files]
+
+    if missing_files:
+        raise Exception(f"missing mandatory files: {', '.join(missing_files)}")
+
+
+def __install_dependencies(
+    manifest: Manifest,
+    app_dir: str,
+    temp_dir: str,
+) -> None:
+    """Install dependencies for the Python app."""
+
+    if manifest.python is None:
+        return
+
+    pip_requirements = manifest.python.pip_requirements
+    if pip_requirements is None or pip_requirements == "":
+        return
+
+    if not os.path.isfile(os.path.join(app_dir, pip_requirements)):
+        raise FileNotFoundError(f"pip requirements file '{pip_requirements}' not found in '{app_dir}'")
+
+    py_cmd = __get_python_command()
+    dep_dir = os.path.join(".nextmv", "python", "deps")
+    command = [
+        py_cmd,
+        "-m",
+        "pip",
+        "install",
+        "-r",
+        pip_requirements,
+        "--platform=manylinux2014_aarch64",
+        "--platform=manylinux_2_17_aarch64",
+        "--platform=manylinux_2_28_aarch64",
+        "--platform=linux_aarch64",
+        "--only-binary=:all:",
+        "--python-version=3.11",
+        "--implementation=cp",
+        "--upgrade",
+        "--no-warn-conflicts",
+        "--target",
+        os.path.join(temp_dir, dep_dir),
+        "--no-input",
+        "--quiet",
+    ]
+    result = subprocess.run(
+        command,
+        cwd=app_dir,
+        text=True,
+        capture_output=True,
+        check=True,
+    )
+    if result.returncode != 0:
+        raise Exception(f"error installing dependencies: {result.stderr}")
+
+
+def __run_command(binary: str, dir: str, redirect_out_err: bool, *arguments: str) -> str:
+    os_agnostic_cmd = binary
+    if arguments:
+        os_agnostic_cmd += " " + " ".join(arguments)
+
+    if platform.system() == "Windows":
+        bin = "cmd"
+        args = ["/c", os_agnostic_cmd]
+    else:
+        bin = "bash"
+        args = ["-c", os_agnostic_cmd]
```

Comment:
This would be awesome to test on both platforms to make sure it works always. :relaxed:

Reply:
Pat did some Windows testing, we _think_ it works. Moving forward.